src/kaleidoscope/key_defs.h

```diff
@@ -225,20 +225,6 @@ typedef kaleidoscope::Key Key_;
 #define SWITCH_TO_KEYMAP           B00000100
 #define IS_CONSUMER                B00001000
 
-/* HID types we need to encode in the key flags for system and consumer control hid controls
-   Each key can only have one, so we don't need to use a bit vector.
-   We need to keep the top two bits clear for defining the keys as synthetic
-   We need to keep the bottom two bits clear for defining the keys as sysctl / consumerctl
-*/
-
-#define HID_TYPE_CL    B00000000
-#define HID_TYPE_LC    B00000100
-#define HID_TYPE_NARY  B00001000
-#define HID_TYPE_OOC   B00001100
-#define HID_TYPE_OSC   B00010000
-#define HID_TYPE_RTC   B00010100
-#define HID_TYPE_SEL   B00011000
```

Comment:
These `HID_TYPE_*` constants were being combined with others in the `flags` byte of both Consumer and System Control `Key` definitions, but this was incorrect. These bits overlap with the `IS_CONSUMER` and `SWITCH_TO_KEYMAP` bits, which are used to detect ConsumerControl and Layer keys, respectively.

For example `HID_TYPE_RTC` would be detected in [`Layer.eventHandler()`](https://github.com/keyboardio/Kaleidoscope/blob/49064d679e187f1ca26a4f15cbed05f4309d7d29/src/kaleidoscope/layers.cpp#L103) as `SWITCH_TO_KEYMAP`, and since all of these `Key` types have the `SYNTHETIC` bit set, this would result in any System or Consumer keys with usage type RTC triggering a call to `Layer.handleKeymapKeyswitchEvent()` (which probably wouldn't do anything, because the `keycode` byte would be out of range, but it's still incorrect). Likewise, any System key with `HID_TYPE_OOC` would be misidentified as a Consumer key in [`handleSyntheticKeyswitchEvent()`](https://github.com/keyboardio/Kaleidoscope/blob/49064d679e187f1ca26a4f15cbed05f4309d7d29/src/kaleidoscope/key_events.cpp#L31), and the System key's HID report would never get sent.

Reply:
That indeed sounds like a bug, but one to be fixed, rather than just throwing away the key type data.

Reply:
I don't think there are enough bits to spare.